packages/experiments/src/components/SelectedItemsList/Items/EditableItem.tsx

```diff
@@ -19,19 +20,15 @@ export type EditableItemProps<T> = {
 // `extends any` to trick the parser into parsing as a type decl instead of a jsx tag
 export const EditableItem = <T extends any>(editableItemProps: EditableItemProps<T>): Item<T> => {
   return React.memo((selectedItemProps: ISelectedItemProps<T>) => {
-    const [isEditing, setIsEditing] = React.useState(false);
-    const setEditingTrue = React.useCallback(() => {
-      setIsEditing(true);
-    }, [setIsEditing]);
-    const setEditingFalse = React.useCallback(() => {
-      setIsEditing(false);
-    }, [setIsEditing]);
+    const { onItemChange, index } = selectedItemProps;
+    const [isEditing, { setTrue: setEditingTrue, setFalse: setEditingFalse }] = useBoolean(false);
+
     const onItemEdited = React.useCallback(
       (_oldItem: T, newItem: T) => {
-        selectedItemProps.onItemChange && selectedItemProps.onItemChange(newItem, selectedItemProps.index);
-        setIsEditing(false);
+        onItemChange && onItemChange(newItem, index);
+        setEditingFalse;
       },
-      [selectedItemProps.onItemChange],
+      [onItemChange, index, setEditingFalse],
```

Comment:
I'm guessing adding a dependency on `selectedItemProps.index` here will be safe but I can revert if desired.

Reply:
Just trying to understand the need for adding index here?

Reply:
The lint rule wants you to exhaustively list all the values the callback depends on, so that a new callback can be generated if one of those values changes. Forgetting to list a dependency value (and therefore keeping a callback with stale values captured from its closure) is a common source of bugs.

The downside of this is that if any of those dependency values change frequently (like if you take callbacks as props, and the component that provides those callbacks re-creates them every render due to not using `useCallback`/similar), your callback also ends up changing frequently, which can cause a chain of effects down the line.

In this case, I'm guessing `index` won't change very frequently, but guessing you'd have a better idea of whether that's actually true based on how it's used in practice. ("Very frequently" only becomes a real problem if it's something that can happen many times per second.)

Reply:
Got it, thanks for the clarification.. index does not change that frequently so should be okay to list here. Sounds good. I am curious then if we need to add setEditingTrue here as well?

Reply:
Not in this case, since it's not used in that particular callback.

Reply:
oh yes you are right , my bad. :)